### PR TITLE
ci(e2e): add plan-only smoke test to CI and Devin E2E testing skill

### PR DESCRIPTION
**Pull request by aaronsteers**
## Summary

Adds a **plan-only CI smoke test** that validates the CI-built provider binary can load, parse HCL, and produce a valid execution plan on every PR. Full E2E testing (apply, drift check, destroy) is handled by a new **Devin skill** that runs against a real Airbyte Cloud workspace.

Also restructures `test-projects/` to separate migration examples from the current generic-only test project, and fixes duplicate CI runs in `test-no-generate.yml`.

### CI Plan Check

**New file:** `.github/workflows/e2e-smoke-test.yml` — reusable workflow (`workflow_call` + `workflow_dispatch`) that:
- Downloads the `provider_binaries` artifact from a specified workflow run (`run-id` is required)
- Configures a `.terraformrc` dev override pointing to the CI-built `linux_amd64` binary
- Runs `terraform init` + `terraform plan` against `test-projects/v1-tf-generic-test/`
- Verifies the compiled binary can resolve data sources (connector registry lookups) and produce a plan

**Modified:** `.github/workflows/test-full.yml` — adds a new `e2e-smoke-test` job that depends on `[check-paths, validate]` and calls the new workflow with `run-id: ${{ github.run_id }}` and `secrets: inherit`. Adds `actions: read` permission. Skips E2E for fork PRs (secrets unavailable).

### Devin E2E Testing Skill

**New file:** `.agents/skills/e2e-testing/SKILL.md` — step-by-step skill for Devin to run the full Terraform lifecycle against Airbyte Cloud:
1. Build the provider binary (`go build -o dist/terraform-provider-airbyte .`)
2. Configure dev override + credentials
3. `terraform plan` → `apply` → `plan -detailed-exitcode` (drift check) → `destroy`
4. Clean up

Uses Devin org secrets (`AIRBYTE_CLOUD_CLIENT_ID`, `AIRBYTE_CLOUD_CLIENT_SECRET`, `AIRBYTE_CLOUD_DEVIN_SANDBOX_WORKSPACE_ID`). Terraform install via `brew install hashicorp/tap/terraform`.

### Test Project Restructuring

**New file:** `test-projects/v1-tf-generic-test/main.tf` — test project using only generic resources (`airbyte_source`, `airbyte_destination`, `airbyte_connection`) and the `airbyte_connector_configuration` data source. Uses `source-faker` and `destination-dev-null` (SILENT mode). Version constraint `>= 1.1.0`.

**Restructured:** `test-projects/` directory:
- `v1-tf-latest-test/` → renamed to `v1-tf-generic-test/` (rewritten to use only generic resources)
- `v0-tf-migration-test/main.tf` → moved to `v0-tf-migration-test/0.x-to-1.0/` (0.x → 1.0 migration reference)
- Old `v1-tf-latest-test/` content → copied to `v0-tf-migration-test/1.0-to-1.1/` (1.0 → 1.1 migration reference)

**Modified:** `test-projects/README.md` — adds project layout table, documents the CI plan check, documents the Devin E2E testing skill, and updated file structure.

### Duplicate CI Fix

**Modified:** `.github/workflows/test-no-generate.yml` — removed the `push` trigger, which was causing every PR commit to run "Test (No Generate)" twice (once for `push`, once for `pull_request`). The `pull_request` trigger alone is sufficient since `test-full.yml` already covers PR validation.

### E2E Test Verification

The full E2E lifecycle was successfully run via the Devin skill against the Devin Sandbox workspace:

| Step | Result |
|---|---|
| `terraform plan` (5 to add) | Passed |
| `terraform apply` (5 added) | Passed |
| **Drift check** (`plan -detailed-exitcode`, exit 0) | **Passed — zero drift** |
| `terraform destroy` (5 destroyed) | Passed |

## Review & Testing Checklist for Human

- [ ] **⚠️ Verify OAuth secrets exist in repo:** The CI workflow uses `secrets.AIRBYTE_CLIENT_ID` / `secrets.AIRBYTE_CLIENT_SECRET` for data source resolution during plan. Check Settings → Secrets and variables → Actions to confirm these exist.
- [ ] **Verify removing `push` trigger from `test-no-generate.yml` is acceptable:** After merge, pushes to `main` will no longer trigger the "Test (No Generate)" workflow. The `test-full.yml` workflow covers PR validation, but if post-merge CI on `main` is desired for this workflow, the `push` trigger should be scoped to `branches: [main]` instead of removed.
- [ ] **Verify `source-faker` definition_id:** The inline approach hardcodes `dfd88b22-b603-4c3d-aad7-3701784586b1` as the Faker source definition ID. Confirm this UUID is current.

**Recommended test plan:**
1. Verify the CI plan-only check passes (check the "E2E Smoke Test" job in the PR checks — currently green)
2. Confirm only one set of "Test (No Generate)" checks appears (no more duplicate push/pull_request runs)
3. Optionally invoke the Devin E2E testing skill to run the full lifecycle — already done once successfully (see above)

### Notes

- The test targets the Devin Sandbox workspace (`266ebdfe-0d7b-4540-9817-de7e4505ba61`), hardcoded as a default in `main.tf`.
- No Go code was changed; `go build ./...` passes cleanly.
- The CI check is plan-only — it does **not** create real resources. Full E2E testing (apply/drift/destroy) is delegated to the Devin skill to avoid needing OAuth secrets in GitHub Actions.
- `run-id` is required for both `workflow_call` and `workflow_dispatch`. For `workflow_call`, `test-full.yml` passes `github.run_id`. For `workflow_dispatch`, the user must supply the run ID of a workflow that produced `provider_binaries`.
- Connection creation can be slow (~25s to 2+ min) — this is expected Airbyte API behavior, not a provider issue.

Link to Devin session: https://app.devin.ai/sessions/2f24c709db8b45ffa42e1331053003ca
Requested by: @aaronsteers

### CODE REVIEW

File: .github/workflows/e2e-smoke-test.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@v3
+        with:
+          terraform_wrapper: false
+
+      - name: Download provider binary artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: provider_binaries
+          path: /tmp/provider-bin
```
Comment (aaronsteers) [inline .github/workflows/e2e-smoke-test.yml:66]: This needs to explicitly get the binary from the current run (if specified in the inputs).

File: .github/workflows/test-full.yml [hunk 2]
```diff
@@ -162,3 +165,10 @@ jobs:
         run: |
           echo "::error::Generated code drift detected. Run /generate on this PR to fix."
           exit 1
+
+  e2e-smoke-test:
+    name: E2E Smoke Test
+    needs: [check-paths, validate]
+    if: needs.check-paths.outputs.should_run == 'true'
+    uses: ./.github/workflows/e2e-smoke-test.yml
+    secrets: inherit
```
Comment (aaronsteers) [inline .github/workflows/test-full.yml:176]: We may need to explicitly pass some kind of reference here as input so it gets artifacts from our current run and not an arbitrary run.

File: .github/workflows/e2e-smoke-test.yml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+          run-id: ${{ inputs.run-id || github.run_id }}
```
Comment (aaronsteers) [inline .github/workflows/e2e-smoke-test.yml]: When not specified, I think any run ID is fine, so coalesce to non/unspecified.

File: .github/workflows/e2e-smoke-test.yml [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+# This verifies that the compiled binary can load, parse HCL, resolve data
+# sources (connector registry lookups), and produce a valid execution plan.
+#
+# For full E2E testing (apply, drift check, destroy), use the Devin E2E
+# testing skill defined in .agents/skills/e2e-testing/SKILL.md.
+#
+# Triggers:
+# - workflow_call: invoked by test-full.yml after provider binaries are built
+# - workflow_dispatch: for manual runs (requires run-id of a workflow that produced provider_binaries)
+
```
[the remark this answers is not in the document]
Comment (aaronsteers) [reply .github/workflows/e2e-smoke-test.yml:19]: When specified as input: get the specified run id. When not specified: use any latest artifact without constraining run id.

File: .github/workflows/e2e-smoke-test.yml [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        with:
+          terraform_wrapper: false
+
+      - name: Download provider binary artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: provider_binaries
+          path: /tmp/provider-bin
+          run-id: ${{ inputs.run-id || '' }}
+          github-token: ${{ inputs.run-id && secrets.GITHUB_TOKEN || '' }}
```
Comment (aaronsteers) [inline .github/workflows/e2e-smoke-test.yml]: You need the GITHUB_TOKEN always. Should not be conditional.

File: .github/workflows/e2e-smoke-test.yml [hunk 6]
```diff
@@ -57,12 +57,21 @@ jobs:
...
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Download provider binary artifact (current run)
+        if: inputs.run-id == ''
         uses: actions/download-artifact@v4
         with:
           name: provider_binaries
           path: /tmp/provider-bin
-          run-id: ${{ inputs.run-id || '' }}
           github-token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (aaronsteers) [inline .github/workflows/e2e-smoke-test.yml]: Let's just change this so that run-id input is required - meaning this workflow can only be called from another source and cannot be run on its own without a run-id input.
